src/Application/Digital/Commands/UpdateEntry.cs

```diff
@@ -0,0 +1,88 @@
+﻿using Application.Common.Exceptions;
+using Application.Common.Interfaces;
+using Application.Common.Models.Dtos.Digital;
+using Application.Common.Models.Operations;
+using AutoMapper;
+using Domain.Entities;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using NodaTime;
+
+namespace Application.Digital.Commands;
+
+public class UpdateEntry
+{
+    public record Command : IRequest<EntryDto>
+    {
+        public Guid CurrentUserId { get; init; }
+        public Guid EntryId { get; init; }
+        public string Name { get; init; } = null!;
+    }
+    
+    public class CommandHandler : IRequestHandler<Command, EntryDto>
+    {
+        private readonly IApplicationDbContext _context;
+        private readonly IMapper _mapper;
+        private readonly IDateTimeProvider _dateTimeProvider;
+
+        public CommandHandler(IApplicationDbContext context, IMapper mapper, IDateTimeProvider dateTimeProvider)
+        {
+            _context = context;
+            _mapper = mapper;
+            _dateTimeProvider = dateTimeProvider;
+        }
+
+        public async Task<EntryDto> Handle(Command request, CancellationToken cancellationToken)
+        {
+            var entry = await _context.Entries
+                .Include(x => x.Owner)
+                .FirstOrDefaultAsync(x => x.Id == request.EntryId, cancellationToken);
+
+            if (entry is null)
+            {
+                throw new KeyNotFoundException("Entry does not exist.");
+            }
+
+            if (entry.Owner.Id != request.CurrentUserId )
+            {
+                throw new UnauthorizedAccessException("User can not access this resource.");
+            }
+            
+            if (entry.OwnerId != request.CurrentUserId)
+            {
+                var permission = await _context.EntryPermissions.FirstOrDefaultAsync(
+                    x => x.EntryId == request.EntryId
+                         && x.EmployeeId == request.CurrentUserId, cancellationToken);
+
+                if (permission is null ||
+                    !permission.AllowedOperations
+                        .Split(",")
+                        .Contains(EntryOperation.Upload.ToString()))
+                {
+                    throw new UnauthorizedAccessException("User cannot access this resource.");
+                }
+            }
```

Comment:
This code kinda smell....
Delete line `46-49`?

Reply:
oh my god, I thought I deleted it, I deserve to die, I will hang myself now